### ISSUE TITLE
New check: Useless setattr / getattr

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#7780]
### Current problem

`getattr` and `setattr` are normally used when getting and setting fields that are not known until runtime. `setattr(thing, field, value)` sets a field of `thing`, but the exact field set depends on the value of the variable `field` (and similar for `getattr(thing, field)`).

This is a very powerful feature of Python. It's great to have if you find yourself in a jam, but it also introduces a lot of instability. It shouldn't be used unless it's really needed.

In particular, it shouldn't be used when the field argument is a string:

```python
setattr(thing, "field", value)

getattr(thing, "field")
```

We don't need to wait until runtime to know which field will be selected, because it says right there. These are equivalent to the simpler:

```python
thing.field = value

thing.field
```

You might think that using a string literal for dynamic field access is a little baroque, and it is. Nonetheless I have seen it in a real-life codebase. I have no idea who wrote it that way or why. My assumption is that it just a strange artifact from a refactor.

### Desired solution

Flag uses of string constants as fields for `setattr` and `getattr`.

**Comment by nickdrozd** [comment]
Two complications:

1. `getattr` with a default value: `getattr(thing, 'field', None)`
2. Objects that implement custom behavior for these functions.
